src/gobupload/relate/__init__.py

```diff
@@ -79,31 +99,29 @@ def _split_job(msg: dict):
 
     assert catalog_name is not None, "A catalog name is required"
 
-    model = GOBModel()
-    catalog = model.get_catalog(catalog_name)
-
+    catalog = gob_model.get(catalog_name)
     assert catalog is not None, f"Invalid catalog name '{catalog_name}'"
 
     if collection_name is None:
-        collection_names = model.get_collection_names(catalog_name)
+        collection_names = catalog['collections'].keys()
     else:
         collection_names = [collection_name]
 
     assert collection_names, f"No collections specified or found for catalog {catalog_name}"
 
     with MessageBrokerConnection(CONNECTION_PARAMS) as connection:
         for collection_name in collection_names:
-            collection = model.get_collection(catalog_name, collection_name)
-            assert collection is not None, f"Invalid collection name '{collection_name}'"
+            try:
+                collection = catalog['collections'][collection_name]
+            except KeyError as exc:
+                raise GOBException(f"Invalid collection name '{collection_name}'") from exc
```

Comment:
Same here. Early validation of the header object can remove the need for this.

Reply:
Done